src/integrationTest/groovy/wooga/gradle/github/AbstractGithubIntegrationSpec.groovy

```diff
@@ -1,132 +1,139 @@
+
+    @Shared
+    @GithubRepository(
+            usernameEnv = "ATLAS_GITHUB_INTEGRATION_USER",
+            tokenEnv = "ATLAS_GITHUB_INTEGRATION_PASSWORD",
+            repositoryPostFixProvider = [TravisBuildNumberPostFix.class],
+            rateLimitHandler = RateLimitHandlerWait.class,
+            resetAfterTestCase = true
+    )
+    Repository testRepo
+
+    def setup() {
+        buildFile << """
+            ${applyPlugin(GithubPlugin)}
+        """.stripIndent()
+    }
+
+    def maybeDelete(String repoName) {
+        try {
+            def repository = client.getRepository(repoName)
+            repository.delete()
+        }
+        catch (Exception e) {
+        }
+    }
+
+    GHContentUpdateResponse createContent(String content, String commitMessage, String path) throws IOException {
+        return testRepo.createContent().content(content).message(commitMessage).path(path).commit();
+    }
+
+    GitHub getClient() {
+        testRepo.client
+    }
+
+    def createRelease(String tagName, String body) {
+        def releaseBuilder = testRepo.createRelease(tagName)
+        releaseBuilder.name(tagName)
+        releaseBuilder.draft(false)
+        releaseBuilder.prerelease(false)
+        releaseBuilder.commitish(testRepo.getDefaultBranch().name)
+        releaseBuilder.body(body)
+        releaseBuilder.create()
+    }
+
+    def createRelease(String tagName) {
+        testRepo.createRelease(tagName,tagName)
+    }
+
+    void cleanupReleases() {
+        testRepo.cleanupReleases()
+    }
+
+    String getTestUserName() {
+        testRepo.userName
+    }
+
+    String getTestUserToken() {
+        testRepo.token
+    }
+
+    String getTestRepositoryName() {
+        testRepo.fullName
+    }
+
+    File createTestAssetsToPublish(int numberOfFiles) {
+        createTestAssetsToPublish(numberOfFiles, null)
+    }
+
+    File createTestAssetsToPublish(int numberOfFiles, String packageName) {
+        File assets = new File(projectDir, "releaseAssets")
+        assets.mkdirs()
+        File packageDirectory = assets
+
+        if (packageName) {
+            packageDirectory = new File(assets, packageName)
+            packageDirectory.mkdirs()
+        }
+
+        numberOfFiles.eachWithIndex { int entry, int i ->
+            createFile("file${i}", packageDirectory) << "test content"
+        }
+
+        assets
+    }
+
+    GHRelease getRelease(String tagName, int retry=10) {
+        def release = (GHRelease) testRepo.listReleases().find { it.tagName == tagName }
+        if(release == null && retry > 0) {
+            sleep(1000)
+            getRelease(tagName, retry-1)
+        }
+        return release
+    }
+
+    GHRelease getReleaseByName(String name) {
+        (GHRelease) testRepo.listReleases().find({ it.name == name })
+    }
+
+    Boolean hasRelease(String tagName) {
+        getRelease(tagName) != null
+    }
+
+    Boolean hasReleaseByName(String name) {
+        getReleaseByName(name) != null
+    }
+}

```

Comment:
what did you actually change here?

Reply:
@Joaquimmnetto : Would you do the honors of updating the description since this implementation was done by you?

Reply:
Actually, I don't think anything was changed here, but idk why git put the diff like this